### PR TITLE
Replace serde[-cbor] with borsh

### PR DESCRIPTION
**Pull request by nhynes**
this constitutes the alpha of v0.3, which will also include a more explicit dispatch step (so it's not yet a release)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,3 +19,8 @@ exclude = ["tests"]
 [profile.release]
 lto = true
 opt-level = 3
+
+[patch.crates-io]
+oasis-std = { path = "oasis-std" }
+oasis-test = { path = "oasis-test" }
+borsh = { git = "https://github.com/nhynes/borsh" }
```
Comment (eauge) [inline Cargo.toml]: why `nhynes/borsh` and not `oasislabs/borsh`?

Comment (nhynes) [reply Cargo.toml]: we can move it to `oasislabs` if you and @peterjgilbert think that's appropriate

Comment (eauge) [reply Cargo.toml]: was just curious if there was a reason. My preference is to have reference a fork in the oasislabs organization

Comment (peterjgilbert) [reply Cargo.toml]: Agreed, we should fork in oasislabs org.

**Comment by rebekahkim** [comment]
How will this affect other services using `serde::Serialize`/`Deserialize` retroactively?

**Comment by nhynes** [comment]
> How will this affect other services using serde::Serialize/Deserialize retroactively?

totally incompatible, but that's captured in the semver bump

File: oasis-build/src/gen/imports.rs [hunk 2]
```diff
@@ -294,6 +309,25 @@ fn get_rustc_version() -> String {
...
+impl HashableChecker {
+    pub fn is_hash(&self) -> bool {
+        !self.contains_nonhashable
+    }
+}
+
+impl oasis_rpc::visitor::IdlVisitor for HashableChecker {
+    fn visit_type(&mut self, ty: &oasis_rpc::Type) {
+        use oasis_rpc::Type::{F32, F64};
+        self.contains_nonhashable |= *ty == F32 || *ty == F64;
```
Comment (peterjgilbert) [inline oasis-build/src/gen/imports.rs]: What's the meaning of non-hashable? And does it include only `F32` and `F64`?

File: tests/Cargo.toml [hunk 3]
```diff
@@ -8,3 +8,6 @@ members = [
 incremental = false
 lto = true
 opt-level = 3
+
+[patch.crates-io]
+borsh = { git = "https://github.com/nhynes/borsh" }
```
Comment (peterjgilbert) [inline tests/Cargo.toml]: Please mirror in oasislabs org.

File: oasis-build/src/gen/imports.rs (hunk 2 above)
Comment (nhynes) [reply oasis-build/src/gen/imports.rs]: it's a type that can't `#[derive(Hash)]`. floats are impermissible because `NaN` is not `PartialEq`

**Pull request merged**